crates/wasi-common/cap-std-sync/src/net.rs

```diff
@@ -230,21 +227,76 @@ macro_rules! wasi_stream_write_impl {
                 Ok(val)
             }
             async fn readable(&self) -> Result<(), Error> {
-                let (readable, _writeable) = self.0.is_read_write()?;
+                let (readable, _writeable) = is_read_write(&self.0)?;
                 if readable {
                     Ok(())
                 } else {
                     Err(Error::io())
                 }
             }
             async fn writable(&self) -> Result<(), Error> {
-                let (_readable, writeable) = self.0.is_read_write()?;
+                let (_readable, writeable) = is_read_write(&self.0)?;
                 if writeable {
                     Ok(())
                 } else {
                     Err(Error::io())
                 }
             }
+
+            async fn sock_recv<'a>(
+                &mut self,
+                ri_data: &mut [std::io::IoSliceMut<'a>],
+                ri_flags: RiFlags,
+            ) -> Result<(u64, RoFlags), Error> {
+                if (ri_flags & !(RiFlags::RECV_PEEK | RiFlags::RECV_WAITALL)) != RiFlags::empty() {
+                    return Err(Error::not_supported());
```

Comment:
The not_supported() function should be better defined to inform the user which feature is not supported, instead of just returning a general error message.

not_supported()函数应该被更好地定义，以告知用户不支持哪个功能，而不是仅返回一般错误消息。